### ISSUE TITLE
Comphet: no gene-like field found in Gene.refGene 

### ISSUE THREAD

**Issue by nroak** [brentp/slivar#41]
I was trying to use [ANNOVAR](http://annovar.openbioinformatics.org/en/latest/) annotated VCF with the Slivar comphet script.
I got the following error:
>[noak@nodecn201  cohort_level]$ ~/resources/slivar/slivar compound-hets \
> -v $family.rare.comphet_int2.vcf \
> -f "Gene.refGene" \
> --sample-field comphet_side \
> -p $family-nosample.ped > $family.rare.comphet.vcf
slivar version: 0.1.6 a135a8703f055b1b9a08632ce0d8887f174e7787
fatal.nim(39)            sysFatal
Error: unhandled exception: comphet.nim(228, 12) `index != -1` [slivar] error! no gene-like field found in Gene.refGene [AssertionError]

I noticed that in the comphet.nim at the corresponding line, it's looking for "SYMBOL" or "GENE" fields. What does this mean?

>  var adesc = ivcf.get_csq_fields(opts.field)
  for check in ["SYMBOL", "GENE"]:
    index = adesc.find(check)
    if index != -1: break
  doAssert index != -1, &"[slivar] error! no gene-like field found in {opts.field}"

**Comment by brentp** [comment]
currently, only bcftools, VEP, and snpEff are supported. Does annovar use sequence ontology terms? if so, it could be supported. any chance you could provide an example VCF? you could even annotate [this one](https://github.com/brentp/slivar/raw/master/tests/ashk-trio.vcf.gz) from the slivar repo.

**Comment by nroak** [comment]
Looks like Annovar terms are mentioned in the [synonyms section](http://www.sequenceontology.org/browser/current_svn/term/SO:0001819) of sequence ontology per their website.
These terms would be under ExonicFunc.refGene tag, gene name will be under Gene.refGene field and so on. Details of this are on [ANNOVAR website](http://annovar.openbioinformatics.org/en/latest/user-guide/gene/#output-file-2-refseq-gene-annotation).

I have [annotated the test VCF](https://github.com/brentp/slivar/files/3498716/ashk-trio.annovar.hg19_multianno.vcf.zip) using annovar, which contains many more fields than what you probably need. The ones below are of interest for gene information.

> ##INFO=<ID=Func.refGene,Number=.,Type=String,Description="Func.refGene annotation provided by ANNOVAR">
> ##INFO=<ID=Gene.refGene,Number=.,Type=String,Description="Gene.refGene annotation provided by ANNOVAR">
> ##INFO=<ID=GeneDetail.refGene,Number=.,Type=String,Description="GeneDetail.refGene annotation provided by ANNOVAR">
> ##INFO=<ID=ExonicFunc.refGene,Number=.,Type=String,Description="ExonicFunc.refGene annotation provided by ANNOVAR">
> ##INFO=<ID=AAChange.refGene,Number=.,Type=String,Description="AAChange.refGene annotation provided by ANNOVAR">

Supp: Annovar script used,
```
$annovar_path/table_annovar.pl $sample.vcf.gz $annovar_path/humandb -buildver hg19 \
-out $sample.annovar -remove \
-protocol refGene,1000g2015aug_all,exac03,exac03nontcga,esp6500siv2_all,gnomad_exome,gnomad_genome,dbnsfp35a,revel,intervar_20180118,dbscsnv11,cosmic70,nci60,clinvar_20180603 \
-operation g,f,f,f,f,f,f,f,f,f,f,f,f,f \
-nastring . -vcfinput >$CURRENTDIR.annovar.log
```

**Comment by brentp** [comment]
hmm. thanks for providing the annotated file. this is different enough from snpEff/bcftools/VEP, that it will be harder to support. bcftools csq command should work; perhaps your VCF has chr prefixes and your gff does not?
I'll leave this issue open and consider better supporting annovar (especially if others chime in expressing the need), but it's not a top priority for me just now.

**Comment by nroak** [comment]
That's what I originally believed, but since bcftools failed, had to put this request in.
bcftools originally threw an error regarding chr, but I now have correct gff3 file. It runs smoothly without any errors and also produces the header line for BCSQ. For some reason it doesn't create the annotation field for my VCF. It runs just okay on the test vcf.

**Comment by brentp** [comment]
hmm. then you could try snpEff (it's quite easy to run), though something is likely odd with your VCF or GFF is this is happening.

**Comment by nroak** [comment]
I agree, just not able to figure out what's odd. Haven't encountered this before.
